core/basic_authorship/block_builder.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_BASIC_AUTHORSHIP_BLOCK_BUILDER_HPP
+#define KAGOME_CORE_BASIC_AUTHORSHIP_BLOCK_BUILDER_HPP
+
+#include "primitives/block.hpp"
+#include "primitives/extrinsic.hpp"
+
+namespace kagome::basic_authorship {
+
+  /**
+   * Creates new blocks from provided header and extrinsics
+   * BlockBuilder collects extrinsics and creates new block and then should be
+   * destroyed
+   */
+  class BlockBuilder {
+   public:
+    virtual ~BlockBuilder() = default;
+
+    /**
+     * Push extrinsic to wait its inclusion to the block
+     * Returns true if xt was pushed, false otherwise
+     */
+    virtual outcome::result<bool> pushExtrinsic(
+        primitives::Extrinsic extrinsic) = 0;
+
+    /**
+     * Create a block from extrinsics and header
+     */
+    [[nodiscard]] virtual primitives::Block bake() const = 0;
```

Comment:
Is it called `bake` for purpose?

Reply:
I think it's from substrate. And actually it's quite a common name for an operation of completion of collecting an object.

Reply:
IMO we should stick to the common naming, and for builder the name is `build`. Also each builder method should return reference to a builder